### PR TITLE
Improve tests' reliability

### PR DESCRIPTION
**Pull request by alfogrillo**
This PR:
- Refactors unit tests removing `Task.sleep` or `Task.yield` whenever is possible with the goal to make tests more reliable.
- Makes `process(viewAction: action)` a sync method
- Fixes integration tests env variables

### CODE REVIEW

File: UnitTests/Sources/Extensions/ViewModelContext.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@testable import ElementX
+
+extension ViewModelContext {
+    @discardableResult
+    func nextViewState() async -> ViewState? {
```
Comment (stefanceriu) [inline UnitTests/Sources/Extensions/ViewModelContext.swift:21]: Can't we put this directly where the `ViewModelContext` is defined? 🤔

File: ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift [hunk 2]
```diff
@@ -32,21 +32,22 @@ class RoomMemberDetailsViewModel: RoomMemberDetailsViewModelType, RoomMemberDeta
...
+            Task { await ignoreUser() }
         case .unignoreConfirmed:
-            await unignoreUser()
+            Task { await unignoreUser() }
         }
     }
 
     // MARK: - Private
 
+    @MainActor
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: Do these really need to be dispatched on the main actor, what happens without it? 🤔

File: UITests/Sources/StartChatScreenUITests.swift [hunk 3]
```diff
@@ -50,7 +50,7 @@ class StartChatScreenUITests: XCTestCase {
     }
     
     func testSearchExactNotExistingMatrixID() {
-        let app = Application.launch(.startChatSearchingInexistingID)
+        let app = Application.launch(.startChatSearchingNonExistingID)
```
Comment (stefanceriu) [inline UITests/Sources/StartChatScreenUITests.swift:53]:
```suggestion
        let app = Application.launch(.startChatSearchingNonexistentID)
```

**Review by stefanceriu — APPROVED** [review]
Such a nice addition, very well done 👏 

Let's also please add publishers to the template vm and coordinator so new screens are done the right way from the start

File: UnitTests/Sources/Extensions/ViewModelContext.swift (hunk 1 above)
Comment (alfogrillo) [reply UnitTests/Sources/Extensions/ViewModelContext.swift:21]: Wanted to put this into an extension just for test targets.
We don't want the EX app to use this stuff right?

File: ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift (hunk 2 above)
Comment (alfogrillo) [reply ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: These methods make side effect on the view state which is supposed to be updated just from the main thread right?

Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: Ah right, good point! 👍

File: UnitTests/Sources/Extensions/ViewModelContext.swift (hunk 1 above)
Comment (stefanceriu) [reply UnitTests/Sources/Extensions/ViewModelContext.swift:21]: Ah right, good point! 👍

File: ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: Right right, but I guess what I'm asking is why they go off the main thread to being with

File: UITests/Sources/StartChatScreenUITests.swift (hunk 3 above)
Comment (alfogrillo) [reply UITests/Sources/StartChatScreenUITests.swift]: Applied

File: ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift (hunk 2 above)
Comment (alfogrillo) [reply ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: They are called like this:
```swift
Task { await ignore() }
```

However I noticed now the whole view model protocol is a main actor:
```swift
@MainActor
protocol RoomDetailsViewModelProtocol
```
So I can delete them. 👍🏻

Comment (stefanceriu) [reply ElementX/Sources/Screens/RoomMemberDetails/RoomMemberDetailsViewModel.swift:50]: Awesome, thanks 👍

**Review by pixlwave — APPROVED** [review]

File: UnitTests/Sources/BugReportViewModelTests.swift [hunk 4]
```diff
@@ -70,19 +68,19 @@ class BugReportViewModelTests: XCTestCase {
...
+        
+        _ = await viewModel
+            .callbackPublisher
+            .filter {
+                guard case .submitFinished = $0 else {
+                    return false
+                }
+                return true
+            }
+            .firstValue
```
[the remark this answers is not in the document]
Comment (alfogrillo) [reply UnitTests/Sources/BugReportViewModelTests.swift]: Like this slightly more.
Added also a convenience method in `AsyncSequence` to get just the first value when needed.
[Here](https://github.com/vector-im/element-x-ios/pull/763/commits/4ec7597b31b2756a456552f1b2189662477758f4) 
Wdyt?

[the remark this answers is not in the document]
Comment (pixlwave) [reply UnitTests/Sources/BugReportViewModelTests.swift]: Oh yeah, that's nice 👍

**Pull request merged**